test/test_foreach.py

```diff
@@ -134,12 +134,25 @@ def _test_binary_op_tensorlists(self, device, dtype, opinfo, N, is_fastpath, dis
             if dtype in torch.testing.get_all_int_dtypes():
                 alpha = 3
             elif dtype.is_complex:
-                dtype = complex(3, 3)
+                alpha = complex(3, 3)
             else:
                 alpha = 3.14
             self._binary_test(dtype, op, ref, inputs, is_fastpath, is_inplace=False, alpha=alpha)
             self._binary_test(dtype, inplace_op, inplace_ref, inputs, is_fastpath, is_inplace=True, alpha=alpha)
 
+        # Tests of implicit broadcasting
+        # When sizes of tensors don't match, foreach functions are supposed to choose slow path.
+        # Thus, argument of `is_fastpath` is set to `False`.
```

Comment:
I'm sorry, where's  `is_fastpath` set to False?

Reply:
my bad, this comment is misleading and kind of out-dated from the code below.
I think I might have wanted to mean "passing False to _binary_test's is_fastpath argument" by this line, but it's actually not.

The logic here is that _binary_test's argument of is_fastpath in general should be False except for both is_fastpath and disable_fastpath are True as done in the code below. 

<!-- A bit of details: In test_foreach.py, we check the `cudaLaunchKernel` of every single foreach function call and the value is compared against n_expected_cudaLaunchKernels which is 1 if not disable_fastpath, N otherwise (N is the number of tensors).
The check of `cudaLaunchKernel` done in ForeachFuncWrapper is `assert cudaLaunchKernel == n_expected_cudaLaunchKernels` if _binary_test's argument of is_fastpath is True, `assert cudaLaunchKernel > n_expected_cudaLaunchKernels` otherwise.

When both is_fastpath and disable_fastpath are true, both cudaLaunchKernel and n_expected_cudaLaunchKernels are N. -->